cdn_lambda/functions/map_to_s3.py

```diff
@@ -0,0 +1,37 @@
+from datetime import datetime
+
+from chexus import Client, TableItem
+
+TABLE_NAME = "project-exodus-prototyping"
+TABLE_REGION = "us-east-2"
+
+
+def lambda_handler(event, context):
+    _ = context
+
+    # Get uri from origin request event
+    request = event["Records"][0]["cf"]["request"]
+
+    client = Client()
+
+    # Get DB table items with matching web_uri key
+    search_results = client.search(
+        item=TableItem(web_uri=request["uri"]),
+        table_name=TABLE_NAME,
+        region=TABLE_REGION
+    )
+
+    items = [TableItem(**i) for i in search_results["Items"]
+             if i.attrs["from_date"] <= datetime.utcnow().date()]
+
+    if items:
+        # Sort the items by date
+        items.sort(key=lambda i: i.from_date)
+
+        # Redirect the request uri
+        request["uri"] = items[-1].attrs["object_key"]
+    else:
+        # Invalid request
+        pass
```

Comment:
I think we need to explicitly give 404 here for reasons:

- if we don't, then users who know about our S3 layout (which is anyone who cares to look, given the code is public) will be able to request objects by checksum.  It's not an intentional feature, so we should block it.
- if the request falls to S3 and generates a 404 there, I think that'll cost us slightly more money, as we still need to pay for the S3 request.

Reply:
Ah, yes, this was a placeholder for the else statement. Yeah, this'll return a 404 response.